modules/gapi/src/compiler/gcompiler.cpp

```diff
@@ -176,6 +176,103 @@ namespace
     }
 } // anonymous namespace
 
+namespace validation
+{
+    class meta_matcher_visitor : cv::util::static_visitor<bool, meta_matcher_visitor>
+    {
+        struct meta_print_visitor : cv::util::static_visitor<bool, meta_print_visitor> {
+            std::ostream& out;
+
+            meta_print_visitor(std::ostream &output) :
+            out (output) {}
+
+            template<class MetaT>
+            bool visit(std::size_t, const MetaT &) {
+                out << cv::detail::meta_to_string<MetaT>();
+                return true;
```

Comment:
btw whats the reason to return `true` here? Is it required by the visitor implementation?

Reply:
Also: I believe our `G*Desc` classes are printable - aren't they? It'd make sense to print out the formats/values themselves in addition to types (also, not very sure if we need to print out the types then)

Reply:
>>btw whats the reason to return true here

currently `specific visitor` must inherit to `utils::static_visitor` which has `return_type` and no limitations here...
Across `return void` is forbidden. But it can be fixed using specialization in simple way (i could do it later)

Another more compilated question is: how we can get return type of Lambda - i think i can cast it to the `std::function` and get type from it. But it requires a more code (and time) ( i also could do it later)

The question is: do you agree for me to finalize/productize this version of visitor? Or it totally should be rejected :)

Reply:
>>Also: I believe our G*Desc classes are printable - aren't they? 

I didn't find such printing by simple search, I'll try search again. But for such kind of check we only compare types, not values. Much information may confuse troubleshooting